pontoon/base/models.py

```diff
@@ -496,6 +515,42 @@ def get_latest_activity(cls, self, extra=None):
 
         return latest_translation.latest_activity if latest_translation else None
 
+    @classmethod
+    def get_chart(cls, self, extra=None):
+        """
+        Get chart for project, locale or combination of both.
+
+        :param self: object to get data for,
+            instance of Projet or Locale
+        :param extra: extra filter to be used,
+            instance of Projet or Locale
+        """
+        chart = None
+
+        if extra is None:
+            chart = cls.get_chart_dict(self)
+
+        else:
+            project = self if isinstance(self, Project) else extra
+            locale = self if isinstance(self, Locale) else extra
+            project_locale = utils.get_object_or_none(ProjectLocale, project=project, locale=locale)
+
+            if project_locale is not None:
+                chart = cls.get_chart_dict(project_locale)
+
+        return chart
+
+    @classmethod
+    def get_chart_dict(cls, obj):
```

Comment:
The data is being JSON serialized in `*_selector.html` templates (needed for the tooltip), which is why I used the `dict`.